src/cli/commands/force/lightning/local/device/list.ts

```diff
@@ -62,29 +62,25 @@ export default class List extends SfdxCommand {
             );
         }
 
-        return new Promise<any>((resolve, reject) => {
-            const deviceList = CommandLineUtils.platformFlagIsIOS(platform)
-                ? this.iOSDeviceList()
-                : this.androidDeviceList();
-
-            resolve(deviceList);
-        });
+        return CommandLineUtils.platformFlagIsIOS(platform)
+            ? this.iOSDeviceList()
+            : this.androidDeviceList();
     }
 
     public async iOSDeviceList(): Promise<IOSSimulatorDevice[]> {
         performance.mark(this.perfMarker.startMarkName);
         const result = await IOSUtils.getSupportedSimulators();
         performance.mark(this.perfMarker.endMarkName);
         this.showDeviceList(result);
-        return result;
+        return Promise.resolve(result);
```

Comment:
This is another pattern that I'm uncertain about—more specifically, for the entirety of this method.  And it boils down to this: functionally/executionally, what's the difference between this:
```typescript
async function asyncThing(): Promise<boolean> {
    const retVal = false;
    return Promise.resolve(retVal);
}
```
And this:
```typescript
async function asyncThing(): Promise<boolean> {
    return new Promise<boolean>((resolve, reject) => {
        const retVal = false;
        resolve(retVal);
    });
}
```
Because my gut says that these will actually exhibit different behaviors—if subtly—at runtime. [This SO answer](https://stackoverflow.com/a/27716590) seems to run through the behavior differences? But I'm still parsing it to understand better what it means at runtime.

Reply:
How I view the two is that both are `then`-able but the second `asyncThing` example allows room to add any logic in the future. As for the real code in `IOSUtils.getSupportedSimulators()` it only returns a resolved value(even when it `catch`es; returns empty array) so simply returning that result in `Promise.resolve` is more compact than the second example, but een simpler, just return `result` as that is equivalent to `Promise.resolve`. I tested that on this branch and it works.

![image](https://user-images.githubusercontent.com/42476999/106199270-49054d80-6159-11eb-9136-48f74f861514.png)

Reply:
That's my takeaway too, after more reading and writing of my own sample code to test the patterns out. 😀 This is probably another one that comes down to context, as to how we choose the code pattern.

Reply:
To back up to the original line of code: I *definitely* support this change, of returning `Promise.resolve(result)` over simply returning `result`. Even if they end up functioning equivalently, I prefer the update to the more intentional code pattern.

Reply:
Another thing I've learned about through this exploration: [the Deferred anti-pattern](https://github.com/petkaantonov/bluebird/wiki/Promise-anti-patterns#the-deferred-anti-pattern). Maybe I'll get good at this someday. 😉